test/jdk/javax/swing/JFileChooser/FileViewNPETest.java

```diff
@@ -0,0 +1,116 @@
+import jdk.test.lib.Platform;
+/*
+ * @test
+ * @bug 6616245
+ * @key headful
+ * @requires (os.family == "windows" | os.family == "linux")
+ * @library /java/awt/regtesthelpers /test/lib
+ * @build PassFailJFrame jdk.test.lib.Platform
+ * @summary Test to check if NPE occurs when using custom FileView.
+ * @run main/manual FileViewNPETest
+ */
+public class FileViewNPETest {
+    static JFrame frame;
+    static JFileChooser jfc;
+    static PassFailJFrame passFailJFrame;
+
+    public static void main(String[] args) throws Exception {
+        SwingUtilities.invokeAndWait(new Runnable() {
+            public void run() {
+                try {
+                    initialize();
+                } catch (InterruptedException | InvocationTargetException e) {
+                    throw new RuntimeException(e);
+                }
+            }
+        });
+        passFailJFrame.awaitAndCheck();
+    }
+
+    static void initialize() throws InterruptedException, InvocationTargetException {
+        //Initialize the components
+        final String INSTRUCTIONS = """
+                Instructions to Test:
+                1. The traversable folder set is "C:\\temp" (In windows)
+                & "/tmp" (In linux), other than these folder all other
+                folders are considered as non-traversable.
+                2. When file chooser appears on screen, select any
+                non-traversable folder (ex-C:/ in windows, / folder in linux)
+                from "Look-In" ComboBox by mouse click/key press. (The folder
+                will not be opened since its non-traversable). Select the same
+                folder again.
+                3. If NPE occurs on second selection then test FAILS, else test
+                is PASS.
+                """;
+        frame = new JFrame("JFileChooser File View NPE test");
+        passFailJFrame = new PassFailJFrame("Test Instructions", INSTRUCTIONS, 5L, 5, 40);
+        jfc = new JFileChooser();
+        String path = "";
+        if (Platform.isWindows()) {
+            path = "C:" + File.separator + "temp";
+        } else {
+            path = File.separator + "tmp";
+        }
+
+        jfc.setCurrentDirectory(new File(path));
+        jfc.setFileView(new CustomFileView(path));
+
+        PassFailJFrame.addTestWindow(frame);
+        PassFailJFrame.positionTestWindow(frame, PassFailJFrame.Position.HORIZONTAL);
+        frame.setDefaultCloseOperation(WindowConstants.EXIT_ON_CLOSE);
+
+        frame.add(jfc, BorderLayout.CENTER);
+        frame.pack();
+        frame.setVisible(true);
+    }
+}
+
+class CustomFileView extends FileView {
+    private String basePath;
+
+    public CustomFileView(String path) {
+        basePath = path;
+    }
+
+    public Boolean isTraversable(File filePath) {
+        if ((filePath != null) && (filePath.isDirectory())) {
+            return filePath.getAbsolutePath().startsWith(basePath);
+        } else {
+            return false;
+        }
```

Comment:
I can be simplified to:
```suggestion
        return ((filePath != null) && (filePath.isDirectory())
                && filePath.getAbsolutePath().startsWith(basePath));
```
I'm not insisting though.

Reply:
Can be done, no issues.

Reply:
Updated.